### PR TITLE
Shop

### PR DESCRIPTION
**Pull request by bythewaters**

### CODE REVIEW

File: app/Customers.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        self.name = name
+        self.shop_name = None
+        self.total_cost1 = {}
+        self.choose_min = []
+
+    def cost_meat(self) -> None:
+        for (name_product1,
+             num_product) in self.customer_product_cart()[self.name].items():
+            price = \
+                self.shops_price()[self.shop_name][name_product1] * num_product
```
Comment (Abnormaltype) [inline app/Customers.py:22]: separate it into the two variables

File: app/Shops.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+class Shop(ShopTrip):
+
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        self.total_cost = 0
+
+    def change_location(self) -> None:
+        self.customer_location()[self.name] \
+            = self.shops_location()[ShopTrip.shop]
```
Comment (Abnormaltype) [inline app/Shops.py:13]: the same

File: app/Shops.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    def purchase(self) -> None:
+        for (name_product1,
+             num_product) in self.customer_product_cart()[self.name].items():
+            price = \
+                self.shops_price()[ShopTrip.shop][name_product1] * num_product
```
Comment (Abnormaltype) [inline app/Shops.py:19]: the same

**Review by Abnormaltype — CHANGES_REQUESTED** [review]

File: app/Customers.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                       - self.customer_location()[self.name][0]) ** 2)
+                     + ((location[1]
+                         - self.customer_location()[self.name][1]) ** 2))
+
+            liter_fuel_to_the_shop = (km_to_the_shop
+                                      * self.cars_customer()[self.name]) / 100
+
+            price_trip_to_the_shop = \
+                liter_fuel_to_the_shop \
+                * self.info["FUEL_PRICE"]
```
Comment (DmytroSvirsa) [inline app/Customers.py:50]: fix code style

**Review by DmytroSvirsa — CHANGES_REQUESTED** [review]

File: app/Customers.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+    def customers_money(self) -> None:
+        for name in self.info["customers"]:
+            if name["name"] == self.name:
+                ShopTrip.money = name["money"]
+                print(f"{self.name} has {name['money']} dollars")
+                break
+
+    def calculate_cost_trip(self) -> None:
+
```
Comment (OleksandrPozden) [inline app/Customers.py:38]: unnecessary indents. remove them

File: app/Customers.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        for name, shop_location in self.shops_location().items():
+            x_coord = self.customer_location()[self.name][0]
+            y_coord = self.customer_location()[self.name][1]
+
```
Comment (OleksandrPozden) [inline app/Customers.py:42]: here

File: app/Customers.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+            km_to_the_shop = sqrt(((shop_location[0] - x_coord) ** 2)
+                                  + ((shop_location[1] - y_coord) ** 2))
+
+            liter_fuel_to_the_shop = (km_to_the_shop
+                                      * self.cars_customer()[self.name]) / 100
+
```
Comment (OleksandrPozden) [inline app/Customers.py:48]: here as well. and so on

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,28 @@
...
+
+
+def shop_trip() -> None:
+    info = Info()
+    for name_customer in info.names_customers():
+
+        shop_trips = ShopTrip(name_customer)
+        purchase_in_shop = Shop(name_customer)
+
+        # Check how much money have customer
```
Comment (OleksandrPozden) [inline app/main.py:13]: remove the comments

**Review by OleksandrPozden — CHANGES_REQUESTED** [review]

**Review by Abnormaltype — APPROVED** [review]
Well done!

**Pull request closed without merging**